src/server/game/Entities/Player/PlayerQuest.cpp

```diff
@@ -1656,30 +1663,37 @@ QuestGiverStatus Player::GetQuestDialogStatus(Object* questgiver)
                 {
                     bool isLowLevel = (getLevel() > (GetQuestLevel(quest) + sWorld->getIntConfig(CONFIG_QUEST_LOW_LEVEL_HIDE_DIFF)));
 
-                    if (quest->IsAutoComplete())
+                    // If quest is auto complete by method type
+                    if (!quest->GetQuestMethod())
                     {
-                        if (isLowLevel)
-                            result2 = DIALOG_STATUS_LOW_LEVEL_REWARD_REP;
-                        else
-                            result2 = DIALOG_STATUS_REWARD_REP;
-                    }
-                    else
-                    {
-                        if (quest->IsDaily())
+                        if (quest->IsRepeatable())
                         {
                             if (isLowLevel)
-                                result2 = DIALOG_STATUS_LOW_LEVEL_AVAILABLE_REP;
+                                result2 = DIALOG_STATUS_LOW_LEVEL_REWARD_REP;
                             else
-                                result2 = DIALOG_STATUS_AVAILABLE_REP;
+                                result2 = DIALOG_STATUS_REWARD_REP;
```

Comment:
what about:

```cpp
result2 = isLowLevel ? DIALOG_STATUS_LOW_LEVEL_REWARD_REP : DIALOG_STATUS_REWARD_REP
```

it's the same thing as your `if/else` but with 1 line of code